### PR TITLE
Dl button

### PR DESCRIPTION
**Pull request by yevgenfs**

### CODE REVIEW

File: Src/lib/UL/spinner.c [hunk 1]
```diff
@@ -38,6 +38,8 @@ static obj_led_t led_arr[] =
         },
 };
 
+static button_t button = {GPIOA, GPIO_PIN_5, 0, cs_NO_PRESS};
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c]: move into driver layer

File: Src/lib/UL/spinner.c [hunk 2]
```diff
@@ -46,8 +48,8 @@ static queue_t queue;
 e_spinner_err_t spinner_init(void)
 {
     return (led_init() == e_led_err_ok
-            && create_queue(&queue, 10) == e_que_err_ok) ?
-            (e_spinner_err_ok) : (e_spinner_err_not_NULL);
+            && create_queue(&queue, 10) == e_que_err_ok && button_init(&button)
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c]: overhead

File: Src/lib/utils/queue.c [hunk 3]
```diff
@@ -22,6 +22,7 @@ queue_err_t create_queue(queue_t *queue_this, uint32_t size)
         {
             return e_que_err_que_allocation_error;
         }
+        return e_que_err_ok;
     }
     return e_que_err_que_NULL_enter;
```
Comment (oleh-plotnikov) [inline Src/lib/utils/queue.c]: e_que_err_que_NULL_enter - more userfriendly enum

File: Src/lib/drivers/button/button.c [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    return e_button_err_callback_already_NULL;
+}
+
+void button_run(void)
+{
+    if (pressed_time_ms >= timeout)
+    {
+        objS_buttonCb(e_event_timeout);
+        pressed_time_ms = 0;
+    }
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.c:65]: empty line

File: Src/lib/drivers/button/button.c [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+button_err_t button_set_timeout(uint32_t timeout_ms)
+{
+    if (timeout_ms != 0)
+    {
+        if (timeout_ms <= MAX_PRESSED_TIME_BORDER)
+        {
+            timeout = timeout_ms;
+            return e_button_err_ok;
+        }
+        return e_button_err_number_more_than_limit_set_to_timeout;
```
Comment (oleh-plotnikov) [inline Src/lib/drivers/button/button.c]: err_invalid_argument

File: Src/lib/UL/spinner.c [hunk 6]
```diff
@@ -42,12 +42,48 @@ static spinner_state_t spinner_state = e_spinner_state_run;
...
 e_spinner_err_t spinner_init(void)
 {
-    return (led_init() == e_led_err_ok
-            && create_queue(&queue, 10) == e_que_err_ok) ?
-            (e_spinner_err_ok) : (e_spinner_err_not_NULL);
+    if(led_init() != e_led_err_ok)
+    {
+        return e_spinner_err_led_not_init;
+    }
+    if(create_queue(&queue, 10) != e_que_err_ok)
```
Comment (oleh-plotnikov) [inline Src/lib/UL/spinner.c:71]: empty line

**Review by oleh-plotnikov — APPROVED** [review]

File: Src/lib/UL/spinner.c (hunk 1 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c]: done

File: Src/lib/UL/spinner.c (hunk 2 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c]: done

File: Src/lib/drivers/button/button.c (hunk 5 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.c]: done

File: Src/lib/UL/spinner.c (hunk 6 above)
Comment (yevgenfs) [reply Src/lib/UL/spinner.c:71]: done

File: Src/lib/drivers/button/button.c (hunk 4 above)
Comment (yevgenfs) [reply Src/lib/drivers/button/button.c:65]: done

**Pull request merged**